myhpom/models/cloudfactory.py

```diff
@@ -0,0 +1,146 @@
+
+import json
+import datetime
+from django.db import models
+
+
+class CloudFactoryRun(models.Model):
+    """Store information about current and past CloudFactory runs."""
+
+    line_id = models.CharField(
+        max_length=64, help_text="the id of the production line at CloudFactory."
+    )
+    run_id = models.CharField(
+        max_length=64,
+        blank=True,
+        default="",
+        help_text="The id of this production run at CloudFactory.",
+    )
+    callback_url = models.CharField(
+        max_length=1024,
+        blank=True,
+        default="",
+        help_text="The URL to which CloudFactory will submit the results of the production run.",
+    )
+    status = models.CharField(
+        max_length=32, blank=True, default="", help_text="The status of the run at CloudFactory."
+    )
+    message = models.TextField(blank=True, help_text="Any message returned by CloudFactory.")
+    created_at = models.DateTimeField(
+        blank=True, null=True, help_text="When the run was created at CloudFactory."
```

Comment:
At present the .created_at field is only used to store the value from CloudFactory, not our own internal creation date. Maybe a new .inserted_at field for that?

Reply:
Part of the defensive programming (IMO anyway, YMMV for other's opinions!) is minimally parsing and storing the internal state of a third party API: if the value is important to the business use case or we need to write SQL queries against it, then parse/store it. Otherwise don't.

Here I think we care about when we made the API request (at creation of this mode instance), not when CF said that they made it.